app/main.py

```diff
@@ -515,7 +528,8 @@ def serve(self):
                 env.action, env.lang, (env.repo, env.charset), get_vars)
             if content is None:
                 response.set_status(404)
-                response.out.write('Not found')
+                page_not_found = 'Page "%s" could not be found' % env.action
```

Comment:
env.action is a kind of technical detail, which doesn't always make sense to end users. Just saying "Not found" sounds enough for me.